com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/security/RateLimiterFilter.java

```diff
@@ -0,0 +1,44 @@
+package com.sap.cloud.lm.sl.cf.web.security;
+
+import com.sap.cloud.lm.sl.cf.web.helpers.RateLimiterProvider;
+import io.github.resilience4j.ratelimiter.RateLimiter;
+import io.github.resilience4j.ratelimiter.internal.AtomicRateLimiter.AtomicRateLimiterMetrics;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.inject.Inject;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class RateLimiterFilter extends OncePerRequestFilter {
+
+    @Inject
+    private RateLimiterProvider rateLimiterProvider;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String ipAddress = request.getRemoteAddr();
+        RateLimiter rateLimiter = rateLimiterProvider.getRateLimiter(ipAddress);
+
+        boolean canProcessRequest = rateLimiter.acquirePermission();
+        AtomicRateLimiterMetrics metrics = (AtomicRateLimiterMetrics) rateLimiter.getMetrics();
+
+        if (canProcessRequest) {
+            response.setHeader("X-Rate-Limit-Remaining", Integer.toString(metrics.getAvailablePermissions()));
+            filterChain.doFilter(request, response);
+            return;
+        }
+        response.setHeader("X-Rate-Limit-Retry-After-Milliseconds",
```

Comment:
Make it consistent with CF and remove this. Add the missing ones. Also extract to constants